packages/tools/kadena-cli/src/utils/globalOptions.ts

```diff
@@ -0,0 +1,330 @@
+  T extends {
+    prompt: any;
+    validation: any;
+    option: Option;
+    expand?: (label: string) => any;
+  },
+>(
+  option: T,
+) => {
+  return (optional: boolean = true) => ({
+    ...option,
+    validation: optional ? option.validation.optional() : option.validation,
+  });
+};
+
+export const globalOptions = {
+  account: createOption({
+    key: 'account' as const,
+    prompt: accountPrompt,
+    validation: z.string(),
+    option: new Option(
+      '-k, --account <account>',
+      'Keypair name',
+    ),
+    expand: async (account: string) => {
+      await ensureAccountsConfiguration();
+      try {
+        return loadAccountConfig(account);
+      } catch (e) {
+        console.log(chalk.yellow(`\nNo account "${account}" found. Please create the account.\n`));
+        await program.parseAsync(['', '', 'account', 'create']);
+        const accountName = await accountPrompt();
+        return loadAccountConfig(accountName);
+      }
+    },
+  }),
+  accountName: createOption({
+    key: 'account' as const,
+    prompt: accountNamePrompt,
+    validation: z.string(),
+    option: new Option(
+      '-a, --account <account>',
+      'Receiver (k:) wallet address',
+    ),
+  }),
+  chainId: createOption({
+    key: 'chainId' as const,
+    prompt: chainIdPrompt,
+    validation: z
+      .string({
+        /* eslint-disable-next-line @typescript-eslint/naming-convention */
+        invalid_type_error: 'Error: -c, --chain-id must be a number',
+      })
+      .min(0)
+      .max(19),
+    option: new Option('-c, --chain-id <chainId>'),
+  }),
+  devnet: createOption({
+    key: 'devnet' as const,
+    prompt: devnetPrompt,
+    validation: z.string(),
+    option: new Option(
+      '-d, --devnet <devnet>',
+      'Devnet name',
+    ),
+    expand: async (devnet: string) => {
+      await ensureDevnetsConfiguration();
+      try {
+        return loadDevnetConfig(devnet);
+      } catch (e) {
+        console.log(chalk.yellow(`\nNo devnet "${devnet}" found. Please create the devnet.\n`));
+        await program.parseAsync(['', '', 'devnet', 'create']);
+        const devnetName = await devnetPrompt();
+        return loadDevnetConfig(devnetName);
+      }
+    },
+  }),
+  devnetName: createOption({
+    key: 'name' as const,
+    prompt: devnetNamePrompt,
+    validation: z.string(),
+    option: new Option('-n, --name <name>', 'Devnet name (e.g. "devnet")'),
+  }),
+  devnetPort: createOption({
+    key: 'port' as const,
+    prompt: devnetPortPrompt,
+    validation: z.number(),
+    option: new Option(
+      '-p, --port <port>',
+      'Port to forward to the Chainweb node API (e.g. 8080)',
+    ).argParser((value) => parseInt(value, 10)),
+  }),
+  devnetUseVolume: createOption({
+    key: 'useVolume' as const,
+    prompt: devnetUseVolumePrompt,
+    validation: z.boolean(),
+    option: new Option(
+      '-u, --useVolume',
+      'Create a persistent volume to mount to the container'
+    ),
+  }),
```

Comment:
Should we parse this into a boolean, or would that interfere with the prompts

`new Option().argParser(Boolean)`